### DISCUSSION TITLE
CLI - Ultralytics YOLOv8 Docs

### DISCUSSION THREAD

**Discussion by giscus** [ultralytics/ultralytics#4140] [Docs]
# CLI - Ultralytics YOLOv8 Docs

Learn how to use Ultralytics YOLO through Command Line: train models, run predictions and exports models to different formats easily using terminal commands.

https://docs.ultralytics.com/usage/cli/

**Comment by RoyalSpite** [comment]
I tried to run prediction cli on Jetson Nano and then say illegal instuctions (core dumped), no prediction at all so how can I fix it thanks. 
ps : I use ultralytics version 8.0.222, python 3.8.12, opencv 4.8.1.78, numpy 1.24.4

**Reply by z00bean** [reply]
Try downgrading NumPy to 1.19.4.
https://elinux.org/Jetson/L4T/TRT_Customized_Example#.22Illegal_instruction_.28core_dumped.29.22

**Reply by pderrenger** [reply]
@z00bean it seems like you're encountering an issue with illegal instructions on your Jetson Nano. This could be related to compatibility issues with the NumPy version you're using. As suggested, you might want to try downgrading NumPy to version 1.19.4, which is known to work well with Jetson platforms.

You can downgrade NumPy using pip with the following command:

```bash
pip install numpy==1.19.4
```

After downgrading, try running the prediction CLI again. If the problem persists, please ensure that all dependencies are compatible with the ARM architecture of the Jetson Nano and that you're using versions of the libraries that are known to work on the Jetson platform.

If you need further assistance, don't hesitate to reach out with more details, and we'll be happy to help! 🛠️

**Comment by omaryomario** [comment]
'yolo' is not recognized as an internal or external command,
operable program or batch file.

**Reply by 20CE034** [reply]
Same error!

**Reply by RizwanMunawar** [reply]
@omaryomario @20CE034 To address the issues, you can update the Ultralytics package or reinstall it using the following commands:

1. Update Command: `pip install -U ultralytics`
2. Uninstall Command: `pip uninstall ultralytics`

This should hopefully resolve the issues.

Thanks
Ultralytics Team!

**Reply by omaryomario** [reply]
it is working now but it gives me another error:



C:\Users\omarb\OneDrive\Desktop\1v1.lol training - the rest of things>yolo task=detect mode=train imgsz=640 data=data.yaml epochs=100 batch=16 name=putyourmodelnamehere
New https://pypi.org/project/ultralytics/8.0.229 available 😃 Update with 'pip install -U ultralytics'
Ultralytics YOLOv8.0.228 🚀 Python-3.11.7 torch-2.1.2+cpu CPU (Intel Core(TM) i5-6500 3.20GHz)
engine\trainer: task=detect, mode=train, model=yolov8n.pt, data=data.yaml, epochs=100, time=None, patience=50, batch=16, imgsz=640, save=True, save_period=-1, cache=False, device=None, workers=8, project=None, name=putyourmodelnamehere, exist_ok=False, pretrained=True, optimizer=auto, verbose=True, seed=0, deterministic=True, single_cls=False, rect=False, cos_lr=False, close_mosaic=10, resume=False, amp=True, fraction=1.0, profile=False, freeze=None, overlap_mask=True, mask_ratio=4, dropout=0.0, val=True, split=val, save_json=False, save_hybrid=False, conf=None, iou=0.7, max_det=300, half=False, dnn=False, plots=True, source=None, vid_stride=1, stream_buffer=False, visualize=False, augment=False, agnostic_nms=False, classes=None, retina_masks=False, show=False, save_frames=False, save_txt=False, save_conf=False, save_crop=False, show_labels=True, show_conf=True, show_boxes=True, line_width=None, format=torchscript, keras=False, optimize=False, int8=False, dynamic=False, simplify=False, opset=None, workspace=4, nms=False, lr0=0.01, lrf=0.01, momentum=0.937, weight_decay=0.0005, warmup_epochs=3.0, warmup_momentum=0.8, warmup_bias_lr=0.1, box=7.5, cls=0.5, dfl=1.5, pose=12.0, kobj=1.0, label_smoothing=0.0, nbs=64, hsv_h=0.015, hsv_s=0.7, hsv_v=0.4, degrees=0.0, translate=0.1, scale=0.5, shear=0.0, perspective=0.0, flipud=0.0, fliplr=0.5, mosaic=1.0, mixup=0.0, copy_paste=0.0, cfg=None, tracker=botsort.yaml, save_dir=runs\detect\putyourmodelnamehere

Dataset 'data.yaml' images not found ⚠️, missing path 'C:\Users\omarb\OneDrive\Desktop\1v1.lol training\images\train\val'
Downloading https://ultralytics.com/assets/coco128.zip to 'C:\Users\omarb\OneDrive\Desktop\1v1.lol training testing\datasets\coco128.zip'...
100%|█████████████████████████████████████████████████████████████████████████████| 6.66M/6.66M [00:01<00:00, 3.85MB/s]
Unzipping C:\Users\omarb\OneDrive\Desktop\1v1.lol training testing\datasets\coco128.zip to C:\Users\omarb\OneDrive\Desk
Dataset download success ✅ (5.8s), saved to C:\Users\omarb\OneDrive\Desktop\1v1.lol training testing\datasets

Overriding model.yaml nc=80 with nc=1

                   from  n    params  module                                       arguments
  0                  -1  1       464  ultralytics.nn.modules.conv.Conv             [3, 16, 3, 2]
  1                  -1  1      4672  ultralytics.nn.modules.conv.Conv             [16, 32, 3, 2]
  2                  -1  1      7360  ultralytics.nn.modules.block.C2f             [32, 32, 1, True]
  3                  -1  1     18560  ultralytics.nn.modules.conv.Conv             [32, 64, 3, 2]
  4                  -1  2     49664  ultralytics.nn.modules.block.C2f             [64, 64, 2, True]
  5                  -1  1     73984  ultralytics.nn.modules.conv.Conv             [64, 128, 3, 2]
  6                  -1  2    197632  ultralytics.nn.modules.block.C2f             [128, 128, 2, True]
  7                  -1  1    295424  ultralytics.nn.modules.conv.Conv             [128, 256, 3, 2]
  8                  -1  1    460288  ultralytics.nn.modules.block.C2f             [256, 256, 1, True]
  9                  -1  1    164608  ultralytics.nn.modules.block.SPPF            [256, 256, 5]
 10                  -1  1         0  torch.nn.modules.upsampling.Upsample         [None, 2, 'nearest']
 11             [-1, 6]  1         0  ultralytics.nn.modules.conv.Concat           [1]
 12                  -1  1    148224  ultralytics.nn.modules.block.C2f             [384, 128, 1]
 13                  -1  1         0  torch.nn.modules.upsampling.Upsample         [None, 2, 'nearest']
 14             [-1, 4]  1         0  ultralytics.nn.modules.conv.Concat           [1]
 15                  -1  1     37248  ultralytics.nn.modules.block.C2f             [192, 64, 1]
 16                  -1  1     36992  ultralytics.nn.modules.conv.Conv             [64, 64, 3, 2]
 17            [-1, 12]  1         0  ultralytics.nn.modules.conv.Concat           [1]
 18                  -1  1    123648  ultralytics.nn.modules.block.C2f             [192, 128, 1]
 19                  -1  1    147712  ultralytics.nn.modules.conv.Conv             [128, 128, 3, 2]
 20             [-1, 9]  1         0  ultralytics.nn.modules.conv.Concat           [1]
 21                  -1  1    493056  ultralytics.nn.modules.block.C2f             [384, 256, 1]
 22        [15, 18, 21]  1    751507  ultralytics.nn.modules.head.Detect           [1, [64, 128, 256]]
Model summary: 225 layers, 3011043 parameters, 3011027 gradients, 8.2 GFLOPs

Transferred 319/355 items from pretrained weights
Freezing layer 'model.22.dfl.conv.weight'
Traceback (most recent call last):
  File "C:\Users\omarb\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\ultralytics\data\base.py", line 115, in get_img_files
    raise FileNotFoundError(f'{self.prefix}{p} does not exist')
FileNotFoundError: train: C:\Users\omarb\OneDrive\Desktop\1v1.lol training\images\train does not exist

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "C:\Users\omarb\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\Scripts\yolo.exe\__main__.py", line 7, in <module>
  File "C:\Users\omarb\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\ultralytics\cfg\__init__.py", line 448, in entrypoint
    getattr(model, mode)(**overrides)  # default args from model
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\omarb\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\ultralytics\engine\model.py", line 338, in train
    self.trainer.train()
  File "C:\Users\omarb\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\ultralytics\engine\trainer.py", line 190, in train
    self._do_train(world_size)
  File "C:\Users\omarb\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\ultralytics\engine\trainer.py", line 290, in _do_train
    self._setup_train(world_size)
  File "C:\Users\omarb\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\ultralytics\engine\trainer.py", line 259, in _setup_train
    self.train_loader = self.get_dataloader(self.trainset, batch_size=batch_size, rank=RANK, mode='train')
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\omarb\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\ultralytics\models\yolo\detect\train.py", line 46, in get_dataloader
    dataset = self.build_dataset(dataset_path, mode, batch_size)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\omarb\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\ultralytics\models\yolo\detect\train.py", line 40, in build_dataset
    return build_yolo_dataset(self.args, img_path, batch, self.data, mode=mode, rect=mode == 'val', stride=gs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\omarb\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\ultralytics\data\build.py", line 80, in build_yolo_dataset
    return YOLODataset(
           ^^^^^^^^^^^^
  File "C:\Users\omarb\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\ultralytics\data\dataset.py", line 41, in __init__
    super().__init__(*args, **kwargs)
  File "C:\Users\omarb\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\ultralytics\data\base.py", line 72, in __init__
    self.im_files = self.get_img_files(self.img_path)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\omarb\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\ultralytics\data\base.py", line 120, in get_img_files
    raise FileNotFoundError(f'{self.prefix}Error loading data from {img_path}\n{HELP_URL}') from e
FileNotFoundError: train: Error loading data from C:\Users\omarb\OneDrive\Desktop\1v1.lol training\images\train
See https://docs.ultralytics.com/datasets/detect for dataset formatting guidance.

**Reply by pderrenger** [reply]
@omaryomario it looks like the dataset path specified in your `data.yaml` file is incorrect or the dataset does not exist at the specified location. The error message indicates that the training images are expected to be found in `C:\Users\omarb\OneDrive\Desktop\1v1.lol training\images\train`, but this directory does not exist.

Please ensure that the path to your dataset is correct in your `data.yaml` file. The `data.yaml` should include the correct paths to your train, val, and (optionally) test image directories, as well as the number of classes and class names. Here's an example structure of what the `data.yaml` file should look like:

```yaml
train: path/to/your/train/images
val: path/to/your/val/images
test: path/to/your/test/images  # optional
nc: number_of_classes
names: ['class1', 'class2', ..., 'classN']
```

Make sure that the paths are correct and that the directories contain your image files and corresponding annotation files. Once you've corrected the paths, try running the training command again. If you continue to have issues, please refer to the documentation for dataset formatting guidance. 📘
